core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -3066,6 +3073,101 @@ public void testCommitStateUnknownNotReconciled() {
         .satisfies(ex -> assertThat(((CommitStateUnknownException) ex).getSuppressed()).isEmpty());
   }
 
+  @Test
+  public void testCustomTableOperationsInjection() throws IOException {
+    AtomicBoolean customTableOpsCalled = new AtomicBoolean();
```

Comment:
Can't we use Mockito.spy to verify that the functions are called instead of introducing booleans for this?

Reply:
I prefer using boolean flags over `Mockito.verify` for this test because:
1. More realistic: The spy approach requires exposing and wrapping the session catalog instance in `newSessionCatalog()`, which isn't how users would actually extend these classes.
2. Test the right thing: The boolean verifies that our custom implementation runs. `newTableOps()` will be called either way (default or custom), so `verify()` doesn't actually prove customization worked.
3. Simpler and straight forward: set a flag in the override to prove it executed. No mocking scaffolding needed. 